subprojects/build-cache-http/src/main/java/org/gradle/caching/http/internal/DefaultHttpBuildCacheServiceFactory.java

```diff
@@ -52,15 +53,21 @@ public BuildCacheService createBuildCacheService(HttpBuildCache configuration, D
         }
         URI noUserInfoUrl = stripUserInfo(url);
 
+        HttpBuildCacheCredentials credentials = configuration.getCredentials();
+        if (!credentialsPresent(credentials) && url.getUserInfo() != null) {
+            credentials = extractCredentialsFromUserInfo(url, credentials);
+        }
+
+        url = noUserInfoUrl;
+
```

Comment:
The mixed use of `url` and `noUserInfoUrl` is confusing. We should not change `url`, and instead use `noUserInfoUrl` below where appropriate.